src/pipboy.cc

```diff
@@ -836,939 +924,1014 @@ static void pipboyDrawText(const char* text, int flags, int color)
 
-            if (gPipboyLinesCount >= 0) {
-                gPipboyCurrentLine = gPipboyLinesCount;
-            }
+         windowRefreshRect(gPipboyWindow, & gPipboyWindowContentRect);
+         inputPauseForTocks(200);
 
-            // Back
-            const char* text1 = getmsg(&gPipboyMessageList, &gPipboyMessageListItem, 201);
-            pipboyDrawText(text1, PIPBOY_TEXT_ALIGNMENT_LEFT_COLUMN_CENTER, _colorTable[992]);
+         if (_view_page <= 0) {
+            pipboyWindowHandleStatus(1024);
+            return;
+         }
 
-            if (gPipboyLinesCount >= 0) {
-                gPipboyCurrentLine = gPipboyLinesCount;
-            }
+         _view_page -= 1;
+      }
 
-            // More
-            const char* text2 = getmsg(&gPipboyMessageList, &gPipboyMessageListItem, 200);
-            pipboyDrawText(text2, PIPBOY_TEXT_ALIGNMENT_RIGHT_COLUMN_CENTER, _colorTable[992]);
+      pipboyRenderHolodiskText();
+      return;
+   }
 
-            windowRefreshRect(gPipboyWindow, &gPipboyWindowContentRect);
-            inputPauseForTocks(200);
+   realIndex = (_view_page_quest * PIPBOY_STATUS_QUEST_LINES) + (a1); // Adjust for pagination
 
-            _view_page -= 1;
+   if (a1 == 1025) {
 
-            if (_view_page < 0) {
-                pipboyWindowHandleStatus(1024);
-                return;
-            }
-        } else {
-            if (gPipboyMouseX > 395) {
-                return;
-            }
+      if (gPipboyMouseX > 395 && gPipboyMouseX < 459 && totalPages > 1) { // prevents click between back/more buttons, except on one page holodisks
+         return;
+      }
 
-            soundPlayFile("ib1p1xx1");
-            blitBufferToBuffer(_pipboyFrmImages[PIPBOY_FRM_BACKGROUND].getData() + PIPBOY_WINDOW_WIDTH * 436 + 254, 350, 20, PIPBOY_WINDOW_WIDTH, gPipboyWindowBuffer + PIPBOY_WINDOW_WIDTH * 436 + 254, PIPBOY_WINDOW_WIDTH);
+      soundPlayFile("ib1p1xx1");
+      handlePipboyPageNavigation(
+         gPipboyMouseX,
+         459, &
+         _view_page_questlist,
+         totalPages,
+         pipboyWindowHandleStatus,
+         []() {
+            // pass -1 to render last chosen location (but now updated with _view_pages_questlist)
+            pipboyWindowQuestList(-1);
+            windowRefreshRect(gPipboyWindow, & gPipboyWindowContentRect);
+         }
+      );
+   }
+
+   if (a1 <= gPipboyQuestLocationsCount) {
+      pipboyWindowQuestList(realIndex);
+   }
+}
 
-            if (gPipboyLinesCount >= 0) {
-                gPipboyCurrentLine = gPipboyLinesCount;
-            }
+static void pipboyWindowQuestList(int a1) {
+
+   // Declare and initialize local variables
+   static int lastLocation = -1; // Store the last location passed, initialized to -1
+   const int maxEntriesPerPage = PIPBOY_STATUS_QUESTLIST_LINES;
+
+   int index = 0;
+   int v13 = 0;
```

Comment:
See above about deobfuscation. Looking at the diff it's hard to tell which code is new and which was just moved around, so it ended up in the diff. So I would treat any code in the diff as new code with some quality standards.